packages/server/internal/services/asset.go

```diff
@@ -0,0 +1,156 @@
+package services
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/alexandros-georgantas/platform-go-challenge/internal/helpers"
+	"github.com/alexandros-georgantas/platform-go-challenge/internal/models"
+	"github.com/alexandros-georgantas/platform-go-challenge/internal/serializers"
+	"gorm.io/gorm"
+)
+
+type AssetService interface {
+	GetAsset(aid uint) (*models.AssetResponse, error)
+	GetAssets(p int, pS int, t string) (serializers.AssetsResponse, error)
+	GetCharts(p int, pS int) (serializers.ChartsResponse, error)
+	GetAudiences(p int, pS int) (serializers.AudienceResponse, error)
+	GetInsights(p int, pS int) (serializers.InsightResponse, error)
+	UpdateDescription(aId uint, d string) (*models.AssetResponse, error)
+}
+
+type assetService struct {
+	db gorm.DB
+}
+
+func NewAssetService(db gorm.DB) (AssetService, error) {
+	return &assetService{db: db}, nil
+}
+
+func (as *assetService) GetAssets(p int, pS int, t string) (serializers.AssetsResponse, error) {
```

Comment:
To be 100% honest the way it is structured now makes it a bit difficult to test. A second iteration would have revealed that :) A proper dependency injection would have been a better solution here. So a dependency for the service would have been its corresponding repository. This approach would allow to mock calls to the corresponding repository/ies and focus on better testing the behavior of service